core/src/mindustry/logic/LExecutor.java

```diff
@@ -1048,6 +1048,55 @@ public static String toString(Object obj){
         }
     }
 
+    public static class FormatI implements LInstruction{
+        public int value;
+
+        public FormatI(int value){
+            this.value = value;
+        }
+
+        FormatI(){}
+
+        @Override
+        public void run(LExecutor exec){
+
+            if(exec.textBuffer.length() >= maxTextBuffer) return;
+
+            int placeholderIndex = -1;
+            int placeholderNumber = 10;
```

Comment:
What is the point of placeholderNumber existing if it is never read from? There is no functional difference between `{0} {1} {2}` and `{5} {6} {7}`, which is very confusing. Why not make it match the log formatting system, and simply replace the single character `@`?

I understand that the system in `bundle.properties` uses positional indexes (`{0}`), but that's because the index actually matters. Here, it doesn't. Most people won't be using this in a world logic context, and will have no idea why the number between the brackets is there.